### DISCUSSION TITLE
In need of coding advice.

### DISCUSSION THREAD

**Discussion by TheBrGuardian** [Vdauphin/HeartsAndMinds#1621] [Question and answer]
So basically I took for myself the goal of shaping the H&M script from simulating MP duty on occupied countries from middle east to a more UN focused Hearts and Minds, simulating the MINUSTAH. With that said, for now, I already "settle up"  the basic that is necessary for Guarantee of Law and Order _(I dared to mess with all the types)_ and now I'm moving to the side missions _(for some reason, because my clan leader likes more to do the Ensure the Safety by kicking the militias out of the cities)_
   I'm now facing a problem with the btc_fnc_findsafepos (BIS_fnc_findSafePos) probably for the map having rivers ("Lingor"),  because some times the Humanitarian tents (btc_side_fnc_supply) are being spawned at the worst places possibles like very near the river or in the middle of the woods. What could I do to solve this problem, not just for this side mission, but for any other use of btc_fnc_findsafepos?

To handle with the Supply side, I had some ideas which weren't scraped during this post and that need to be coded: 
- Make a cleared town as the center of the SafePos, plus making the parameters more solid or checking for proximity to roads;
[maybe will have to recheck city to city and in false re-roll the side poll]

- Remove the composition and make it a supply delivery mission, just like the real H&M strategy;
[I came up with this one now and is very simple, in a way that I could scavenge the checkpoint script and make it chose roads from safe towns]

PS: I'm not sure if this adaptation will ever be shared since It's already in a point were I'm questioning the cost/benefit of making it easier to port, because I kinda hyperfocused and sacrificed the time I was supposed to be saving my grades on university.
_Who dares wins_

**Comment by Vdauphin** [comment]
Hello,

> What could I do to solve this problem, not just for this side mission, but for any other use of btc_fnc_findsafepos?

[`btc_fnc_findsafepos` use `BIS_fnc_findSafePos` behind](https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/common/findsafepos.sqf#L37). You can tweak as you want (see https://community.bistudio.com/wiki/BIS_fnc_findSafePos). You can also test quickly those by using something like that: `player setPos ([getPos player] call btc_fnc_findsafepos;)`

In supply : https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/side/supply.sqf#L35C76-L35C95 the `false mean that water is not allowed but for some reason `BIS_fnc_findSafePos` still return on water
May be `surfaceIsWater` return false

If it spawn inside tree it is may be just the BIS_fnc_findSafePos failing to find a suitable location

One thing can badly affect searching is if `"cachingRadius"` is too small. You can see if it is the issue by enable debug and look at `cachingRadius` by defining [the player spawn radius to 0](https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/def/param.hpp#L410-L414). Then you will need to see if it does include the suitable place

Good luck and have fun
